crates/holochain_zome_types/src/op.rs

```diff
@@ -0,0 +1,291 @@
+        header: SignedHashed<EntryCreationHeader>,
+        /// The new [`Entry`] to store.
+        entry: Entry,
+    },
+    /// Registers an update from an instance of an [`Entry`] in the DHT.
+    /// This is the act of creating a [`Header::Update`] and
+    /// publishing it to the DHT.
+    /// Note that the [`Header::Update`] stores an new instance
+    /// of an [`Entry`] and registers it as an update to the original [`Entry`].
+    /// This operation is only concerned with registering the update.
+    RegisterUpdate {
+        /// The signed and hashed [`Header::Update`] that registers the update.
+        update: SignedHashed<Update>,
+        /// The new [`Entry`] that is being updated to.
+        new_entry: Entry,
+        /// The original [`EntryCreationHeader`] that created
+        /// the original [`Entry`].
+        /// Note that the update points to a specific instance of the
+        /// of the original [`Entry`].
+        original_header: EntryCreationHeader,
+        /// The original [`Entry`] that is being updated from.
+        original_entry: Entry,
+    },
+    /// Registers a deletion of an instance of an [`Entry`] in the DHT.
+    /// This is the act of creating a [`Header::Delete`] and
+    /// publishing it to the DHT.
+    RegisterDelete {
+        /// The signed and hashed [`Header::Delete`] that registers the deletion.
+        delete: SignedHashed<Delete>,
+        /// The original [`EntryCreationHeader`] that created
+        /// the original [`Entry`].
+        original_header: EntryCreationHeader,
+        /// The original [`Entry`] that is being deleted.
+        original_entry: Entry,
+    },
+    /// Registers a new [`Header`] on an agent source chain.
+    /// This is the act of creating any [`Header`] and
+    /// publishing it to the DHT.
+    RegisterAgentActivity {
+        /// The signed and hashed [`Header`] that is being registered.
+        header: SignedHeaderHashed,
+    },
+    /// Registers a link between two [`Entry`]s.
+    /// This is the act of creating a [`Header::CreateLink`] and
+    /// publishing it to the DHT.
+    /// The authority is the entry authority for the base [`Entry`].
+    RegisterCreateLink {
+        /// The signed and hashed [`Header::CreateLink`] that registers the link.
+        create_link: SignedHashed<CreateLink>,
+        /// The base [`Entry`] that is being linked from.
+        base: Entry,
+        /// The target [`Entry`] that is being linked to.
+        target: Entry,
+    },
+    /// Deletes a link between two [`Entry`]s.
+    /// This is the act of creating a [`Header::DeleteLink`] and
+    /// publishing it to the DHT.
+    /// The delete always references a specific [`Header::CreateLink`].
+    RegisterDeleteLink {
+        /// The signed and hashed [`Header::DeleteLink`] that registers the deletion.
+        delete_link: SignedHashed<DeleteLink>,
+        /// The link that is being deleted.
+        create_link: CreateLink,
```

Comment:
looking at the things in here that are `SignedHashed<..>` and that are not... Do we even need to pass the signatures to WASM? aren't all the signatures validated in the sys_validation?

Reply:
Yeh this is sort of an open question. They are validated in sys validation but we do already pass the signatures inside the `Element` so this is consistent with that.